crates/distribution-types/src/requirements.rs

```diff
@@ -0,0 +1,35 @@
+use pep508_rs::Requirement;
+use uv_normalize::ExtraName;
+
+/// A set of requirements as requested by a parent requirement.
+///
+/// For example, given `flask[dotenv]`, the `RequestedRequirements` would include the `dotenv`
+/// extra, along with all of the requirements that are included in the `flask` distribution
+/// including their unevaluated markers.
+#[derive(Debug, Clone)]
+pub struct RequestedRequirements {
```

Comment:
Why here instead of `uv-requirements` or `uv-types`?

Reply:
`uv-types` could make sense. `uv-requirements` is a heavy dependency for (e.g.) the resolver to pull in.